tests/server/integration/smoke.js

```diff
@@ -32,33 +32,36 @@ describe('Smoke', function () {
 
 		// wait for the servers to spin up, then hit the home page
 		var serversReadyPromise = serversUnderTest.readyForTest();
-		serversReadyPromise
+		return serversReadyPromise
 			.then(function () {
 				var getHomePromise = getPath('/home');
-				getHomePromise
+				return getHomePromise
 					.then(function (data) {
 						getHomeResponseBody = data;
 						var regexp = /\/public\/resources\/main-.+\.css/;
 						var found = getHomeResponseBody.match(regexp)[0];
 						var getCSSPromise = getPath(found);
-						getCSSPromise
+						return getCSSPromise
 							.then(function (data) {
 								getCSSResponseBody = data;
-								serversUnderTest.finish();
-								done(); // tell mocha it can now run the suite
+								return serversUnderTest.finish();
 							})
 							.catch(function (err) {
 								console.error(err);
-								serversUnderTest.finish();
+								return serversUnderTest.finish();
 							});
 					})
 					.catch(function (err) {
 						console.error(err);
-						serversUnderTest.finish();
+						return serversUnderTest.finish();
 					});
 			});
 	}); // before
 
+	after(function () {
```

Comment:
Except for this addition, I am struggling to understand why the other changes have been made in this test?

Reply:
This test was not handling error cases in the nested promise chain, leading to the test output giving misleading output about what's failed. One option would have been to call `done(err)` in the nested promises in a continuation-passing style, but given that this is a test based on promises I felt it more appropriate to return the rejected Promise on failure, or the resolved Promise on success.

Happy to go back to the `done` injection, but then we'd need to handle the errors on each of the promises.

Reply:
OK makes sense, thanks!